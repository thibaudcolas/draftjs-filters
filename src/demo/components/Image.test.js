import React from "react"
import { shallow } from "enzyme"
import { convertFromRaw } from "draft-js"

import Image from "./Image"

describe("Image", () => {
  it("renders", () => {
    const content = convertFromRaw({
      entityMap: {
        "0": {
          type: "IMAGE",
          data: {
            src: "/example.png",
          },
        },
      },
      blocks: [
        {
          key: "a",
          text: " ",
          entityRanges: [
            {
              offset: 0,
              length: 1,
              key: 0,
            },
          ],
        },
      ],
    })

    expect(
      shallow(<Image contentState={content} block={content.getFirstBlock()} />),
    ).toMatchSnapshot()
  })

  it("no entity", () => {
    const content = convertFromRaw({
      entityMap: {},
      blocks: [
        {
          key: "a",
          text: " ",
        },
      ],
    })

    expect(
      shallow(<Image contentState={content} block={content.getFirstBlock()} />),
    ).toMatchSnapshot()
  })
})
