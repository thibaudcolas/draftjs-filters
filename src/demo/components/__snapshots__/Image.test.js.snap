// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image no entity 1`] = `
<img
  alt=""
  src="/404"
  width="256"
/>
`;

exports[`Image renders 1`] = `
<img
  alt=""
  src="/example.png"
  width="256"
/>
`;
